src/state/reducer.js

```diff
@@ -339,6 +339,23 @@ export default (state: State = clean('IDLE'), action: Action): State => {
     });
   }
 
+  if (action.type === 'UPDATE_DROPPABLE_IS_ENABLED') {
+    const { id, isEnabled } = action.payload;
+    return {
+      ...state,
+      dimension: {
+        ...state.dimension,
+        droppable: {
+          ...state.dimension.droppable,
+          [id]: {
```

Comment:
rather than going this deep can we pull the dimension out?
Also, some error checking would be good also

```js
const target = state.dimension.droppable[id]

if(!target) {
  console.error('cannot update enabled flag on droppable that does not have a dimension');
  return clean();
}

if(target.isEnabled === isEnabled) {
  console.warn(`trying to set droppable isEnabled to ${isEnabled}` but it is already ${isEnabled}`);
  return state;
}

const updated = {
 ...target,
 isEnabled,
};

return {
       ...state,
       dimension: {
         ...state.dimension,
         droppable: {
           ...state.dimension.droppable,
           [id]: updated,
      }
 }
```